src/modules/rents/rents.service.ts

```diff
@@ -0,0 +1,65 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { DatabaseService } from 'src/database/database.service';
+import { Rent } from 'src/entities/rent.entity';
+import { checkWeekDay } from 'src/helpers/date.helper';
+import { calculateRentPrice } from 'src/helpers/price.helper';
+import { CreateRentDto } from './dto/add-rent.dto';
+import { CarsService } from '../cars/cars.service';
+
+@Injectable()
+export class RentsService {
+  constructor(
+    private databaseService: DatabaseService,
+    private carsService: CarsService,
+  ) {}
+  async getAllRents(): Promise<Rent[]> {
+    const res = this.databaseService.executeQuery(`SELECT * FROM rents`);
+    return res;
+  }
+
+  async rentCar(
+    car_id: number,
+    createRentDto: CreateRentDto,
+  ): Promise<Rent[] | string> {
+    const { start, end } = createRentDto;
+    const price = calculateRentPrice(start, end);
+    const isAvailable = await this.carsService.checkAvailability(
+      car_id,
+      createRentDto,
+    );
+
+    if (isAvailable === 'NO')
+      throw new BadRequestException(
+        'Car is not available at requested time period',
+      );
+    if (isAvailable === 'ID does not exist.')
+      throw new BadRequestException('Car with such ID does not exist.');
+    if (typeof price !== 'number') {
+      throw new BadRequestException(price);
+    }
+    if (checkWeekDay(start, end)) {
+      throw new BadRequestException(
+        'Renting start and end dates cannot land on Saturday or Sunday.',
+      );
+    }
```

Comment:
would be better to move to the top of the method 
because if dates are wrong you can avoid request to database